bpf/bpf_sock.c

```diff
@@ -256,31 +266,26 @@ sock4_wildcard_lookup_full(struct lb4_key *key __maybe_unused,
  * service translation.
  */
 static __always_inline bool
-sock4_skip_xlate_if_same_netns(struct bpf_sock_addr *ctx __maybe_unused,
-			       const struct lb4_backend *backend __maybe_unused)
+sock4_skip_xlate_from_ctx_to_svc(struct bpf_sock_addr *ctx __maybe_unused,
+				 __be32 address __maybe_unused, __be16 port __maybe_unused)
 {
-	struct bpf_sock_tuple tuple = {
-		.ipv4.daddr = backend->address,
-		.ipv4.dport = backend->port,
-	};
-	struct bpf_sock *sk = NULL;
-
-	switch (ctx->protocol) {
-	case IPPROTO_TCP:
-		sk = sk_lookup_tcp(ctx, &tuple, sizeof(tuple.ipv4),
-				   BPF_F_CURRENT_NETNS, 0);
-		break;
-	case IPPROTO_UDP:
-		sk = sk_lookup_udp(ctx, &tuple, sizeof(tuple.ipv4),
-				   BPF_F_CURRENT_NETNS, 0);
-		break;
-	}
-
-	if (sk) {
-		sk_release(sk);
+#ifdef HAVE_NETNS_COOKIE
+	__net_cookie cookie = get_netns_cookie(ctx);
+	struct skip_lb4_key key;
+	__u8 *val = NULL;
+
+	memset(&key, 0, sizeof(key));
+	key.netns_cookie = cookie;
+	key.address = address;
+	key.port = port;
+	printk("bpf_sock-trace: netns %llu %u\n", key.netns_cookie, key.address);
+	val = map_lookup_elem(&LB4_SKIP_MAP, &key);
```

Comment:
Seems like we weren't doing any bpf map lookups in this code-path prior to this, is there any performance concerns here?

Reply:
The old logic was invoking `sk_lookup_` helper in the kernel, which looks up a socket in the kernel hash table. While the new logic works differently, the performance concerns are not significant as the lookup is done only once at the socket `connect` calls.